src/vtvalidate.cpp

```diff
@@ -118,23 +145,21 @@ NAN_METHOD(isValid) {
     }
     v8::Local<v8::Function> callback = info[1].As<v8::Function>();
 
-    // Check first argument, should be an 'options' object
-    if (!info[0]->IsObject()) {
-        return utils::CallbackError("first arg 'options' must be an object",
-                                    callback);
-    }
-    v8::Local<v8::Object> options = info[0].As<v8::Object>();
-
-    // Check options object for the "louder" property, which should be a boolean
-    // value
-    if (options->Has(Nan::New("louder").ToLocalChecked())) {
-        v8::Local<v8::Value> louder_val =
-            options->Get(Nan::New("louder").ToLocalChecked());
-        if (!louder_val->IsBoolean()) {
-            return utils::CallbackError("option 'louder' must be a boolean",
-                                        callback);
-        }
-        louder = louder_val->BooleanValue();
+    // BUFFER: check first argument, should be a pbf object
+    v8::Local<v8::Value> buffer_val = info[0];
+    if (buffer_val->IsNull() ||
+        buffer_val->IsUndefined()) {
+        utils::CallbackError("first arg is empty", callback);
+        return;
+    }
+
+    v8::Local<v8::Object> buffer = buffer_val->ToObject();
+
+    if (buffer->IsNull() ||
+        buffer->IsUndefined() ||
+        !node::Buffer::HasInstance(buffer)) {
+        utils::CallbackError("first arg 'buffer' must be a Protobuf object", callback);
```

Comment:
Should this be a callback error? If the method is used improperly prior to a callback, shouldn't the error appear there? /cc @springmeyer

Reply:
@flippmoke are you asking if this should be a `throw` instead of returning the error in the callback? In general I think this is fine: returning in the callback is preferable most cases (though this does differ from node-mapnik were we throw - but that is because I never took the time to figure out how to return in the callback without hitting the threadpool).